tests/benchmark/crud_test.go

```diff
@@ -0,0 +1,920 @@
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package benchmark
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/apigee/registry/connection"
+	"github.com/apigee/registry/rpc"
+	"google.golang.org/genproto/protobuf/field_mask"
+)
+
+func getRootResource(b *testing.B) string {
+	b.Helper()
+
+	p, ok := os.LookupEnv("REGISTRY_PROJECT_IDENTIFIER")
+	if !ok || p == "" {
+		p = "bench"
+	}
+	return fmt.Sprintf("projects/%s/locations/global", p)
+}
+
+func loadVersionCount(b *testing.B) int {
+	b.Helper()
+
+	s, ok := os.LookupEnv("REGISTRY_VERSION_COUNT")
+	if !ok {
+		return 1
+	}
+
+	c, err := strconv.Atoi(s)
+	if err != nil {
+		b.Fatalf("Setup: Invalid REGISTRY_VERSION_COUNT %q: count must be an integer", s)
+	}
+
+	return c
+}
+
+func getApiName(i int, b *testing.B) string {
+	b.Helper()
+
+	prefix, ok := os.LookupEnv("API_NAME_PREFIX")
+	if !ok {
+		prefix = "benchtest"
+	}
+	s, ok := os.LookupEnv("API_NAME_START_OFFSET")
+	if ok {
+		offset, err := strconv.Atoi(s)
+		if err != nil {
+			b.Fatalf("Setup: Invalid API_NAME_START_OFFSET %q: offset must be an integer", s)
+		}
+		i += offset
+	}
+
+	return fmt.Sprintf("%s-%d", prefix, i)
+}
+
+func readAndGZipFile(filename string) (*bytes.Buffer, error) {
+	fileBytes, _ := ioutil.ReadFile(filename)
+	var buf bytes.Buffer
+	zw, _ := gzip.NewWriterLevel(&buf, gzip.BestCompression)
+	_, err := zw.Write(fileBytes)
+	if err != nil {
+		return nil, err
+	}
+	if err := zw.Close(); err != nil {
+		return nil, err
+	}
+	return &buf, nil
+}
+func pauseAfter1000Iterations(i int) {
+	if i%1000 == 0 {
+		/*
+			Pause 5 seconds every 1000 iterations to reduce chances of failure on
+			CREATE, UPDATE, DELETE
+		*/
+		time.Sleep(5 * time.Second)
+	}
+}
+
+func BenchmarkCreateApi(b *testing.B) {
+
+	rootResource := getRootResource(b)
+	ctx := context.Background()
+	gapicClient, err := connection.NewClient(ctx)
+	if err != nil {
+		b.Fatalf("Setup: Failed to create client: %s", err)
+	}
+	client := gapicClient.GrpcClient()
+
+	for i := 1; i <= b.N; i++ {
+		pauseAfter1000Iterations(i)
+
+		apiId := getApiName(i, b)
+
+		req := &rpc.CreateApiRequest{
+			Parent: rootResource,
+			ApiId:  apiId,
+			Api: &rpc.Api{
+				DisplayName: apiId,
+				Description: fmt.Sprintf("Description for %s", apiId),
+			},
+		}
+		b.StartTimer()
+		_, err := client.CreateApi(ctx, req)
+		b.StopTimer()
+		/**
+		Go Benchmarking always runs all the benchmark tests once before it runs
+		for the duration specified. So the first row always returns error.
+		*/
```

Comment:
Do you have a link to any documentation about this? Why does it cause an error?

Reply:
Added this info in the commit message: 

https://github.com/golang/go/issues/32051
The issue seems to be fixed for --benchtime=1x but does not fix --benchtime=100X